src/utils/httpVariableProviders/environmentVariableProvider.ts

```diff
@@ -49,6 +50,27 @@ export class EnvironmentVariableProvider implements HttpVariableProvider {
         const variables = this._settings.environmentVariables;
         const currentEnvironmentVariables = variables[environmentName];
         const sharedEnvironmentVariables = variables[EnvironmentController.sharedEnvironmentName];
+        this.mapEnvironmentVariables(currentEnvironmentVariables, sharedEnvironmentVariables);
         return Object.assign({}, sharedEnvironmentVariables, currentEnvironmentVariables);
     }
+
+    private mapEnvironmentVariables(current: any, shared: any) {
+        for (let key in current) {
```

Comment:
minor: Use for...of together with `Object.entries` since you need to use both key and value pair like following:
```js
for (const [key, value] of Object.entries(current)) {
    // Logic
}
```